### PR TITLE
[IMP] event: convert field IconSelector

### PR DESCRIPTION
**Pull request by mcm-odoo**
This commit rewrites IconSelectorField using the new framework.

### CODE REVIEW

File: addons/event/static/src/icon_selection_field/icon_selection_field.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        return this.props.value.charAt(0).toUpperCase() + this.props.value.slice(1);
+    }
+}
+IconSelectionField.template = "event.IconSelectionField";
+IconSelectionField.props = {
+    ...standardFieldProps,
+    icons: Object,
+};
+
+IconSelectionField.displayName = _lt("Integer");
```
Comment (aab-odoo) [inline addons/event/static/src/icon_selection_field/icon_selection_field.js:23]: ?

**Review by aab-odoo — APPROVED** [review]
Same question here as in  the holidays pr, are you sure we can remove the legacy code?

**Comment by mcm-odoo** [comment]
@aab-odoo the registry key and the module name are not referenced anywhere else

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**